pagerduty/resource_pagerduty_event_orchestration_path_service.go

```diff
@@ -213,27 +176,30 @@ func resourcePagerDutyEventOrchestrationPathService() *schema.Resource {
 	}
 }
 
-func resourcePagerDutyEventOrchestrationPathServiceRead(d *schema.ResourceData, meta interface{}) error {
+func resourcePagerDutyEventOrchestrationPathServiceRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	client, err := meta.(*Config).Client()
 	if err != nil {
-		return err
+		return diag.FromErr(err)
 	}
 
 	id := d.Id()
 	var path *pagerduty.EventOrchestrationPath
 	retryErr := resource.Retry(2*time.Minute, func() *resource.RetryError {
 		t := "service"
-		log.Printf("[INFO] Reading PagerDuty Event Orchestration Path of type %s for orchestration: %s", t, id)
+		log.Printf("[INFO] Reading PagerDuty Event Orchestration Path of type %s for service: %s", t, id)
 
 		path, _, err = client.EventOrchestrationPaths.Get(id, t)
+
 		if err != nil {
 			time.Sleep(2 * time.Second)
 			return resource.RetryableError(err)
 		}
+
 		return nil
 	})
+
 	if retryErr != nil {
-		return retryErr
+		return diag.FromErr(retryErr)
```

Comment:
Also add context-aware methods for status get/update endpoint

Reply:
Addressed in https://github.com/PagerDuty/terraform-provider-pagerduty/pull/618/commits/9c7ddd0faf4d463f7dd1118d14936863a7232114